lib/engine/game/g_1817/game.rb

```diff
@@ -978,6 +978,8 @@ def operating_round(round_num)
         end
 
         def or_round_finished
+          return unless @depot.upcoming.any?
```

Comment:
Don't call `Enumerable.any?` (or `Enumerable.none?`) without a block, this method will iterate over each element in the array testing if it is nil. Use `Array.empty?` instead 
```suggestion
          return if @depot.upcoming.empty?
```

Reply:
Agree with this change. `any?` is significantly more expensive than `.empty?`.